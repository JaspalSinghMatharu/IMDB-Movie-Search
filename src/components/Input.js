import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    /** value of the input feild */
    value: PropTypes.string,
    /** Placeholder text for input feild */
    placeholder: PropTypes.string,
    /** Label for the input feild */
    label: PropTypes.string,
    /** id for the input */
    id: PropTypes.string,
    /** A function to be called on input change */
    handleChange: PropTypes.func,
}

const defaultProps = {
    value: '',
    placeholder: 'Select',
    label: "",
}

const Input = (props) => {
    const {label, id, value, placeholder, handleChange} = props;
    return (
        <div className="form-item">
            {label && <label htmlFor={id} className="form-label">{label}</label>}
            <input className="form-input" placeholder={placeholder} value={value} id={id} onChange={handleChange} />
        </div>
    )
}

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default Input;