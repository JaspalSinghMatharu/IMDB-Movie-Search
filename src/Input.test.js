import { render, screen } from '@testing-library/react';
import Input from './components/Input';

test('render Input component', () => {
  const {wrapper} = render(<Input />);
  expect(wrapper).toMatchSnapshot();
});
