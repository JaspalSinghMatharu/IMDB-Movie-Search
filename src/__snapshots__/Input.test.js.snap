// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AutoComplete component 1`] = `undefined`;

exports[`render Input component 1`] = `undefined`;
